segmentation faults when parsing Opentitan system Verilog top files

When parsing the system Verilog files in the Makefile.in 
[Makefile.in.txt](https://github.com/alainmarcel/Surelog/files/6419448/Makefile.in.txt)

 using this command line: 
-parse -sverilog -d coveruhdm ${SURELOG_FLAGS} $(INCLUDE) $(TOP_FILE).
 where
SURELOG_FLAGS := --disable-feature=parametersubstitution +define+SYNTHESIS
INCLUDE := -I$(EARLGRAY_BUILD)/src/lowrisc_prim_assert_0.1/rtl \
        -I$(EARLGRAY_BUILD)/src/lowrisc_prim_util_memload_0/rtl \

I'm running into segmentation faults. a screen shoot contains frame that throw the segmentation fault  
![frame](https://user-images.githubusercontent.com/60285497/116969866-47e33f80-acb7-11eb-957d-366326d5388f.PNG)

Please attach the surelog.log file to this issue. I'll get the complete command line without having to debug your Makefile.

alain@alain-i7:~/Surelog/third_party/tests/Titan/issue$ make 
Makefile:17: tests/opentitan/Makefile.in: No such file or directory
make: *** No rule to make target 'tests/opentitan/Makefile.in'.  Stop.
alain@alain-i7:~/Surelog/third_party/tests/Titan/issue$ find . -name Makefile.in
alain@alain-i7:~/Surelog/third_party/tests/Titan/issue$

--- a/src/DesignCompile/CompileType.cpp
+++ b/src/DesignCompile/CompileType.cpp
@@ -1090,20 +1090,26 @@ UHDM::typespec* CompileHelper::compileTypespec(
         }
       }
       if (!result) {
-        UHDM::VectorOfparam_assign* param_assigns = component->getParam_assigns();
-        if (param_assigns) {
-          for (param_assign* param : *param_assigns) {
-            const std::string& param_name = param->Lhs()->VpiName();
-            if (param_name == typeName) {
-              const any* rhs = param->Rhs();
-              if (const expr* exp = dynamic_cast<const expr*>(rhs)) {
-                int_typespec* its = buildIntTypespec(compileDesign, param->VpiFile(), typeName, exp->VpiValue(), param->VpiLineNo(), param->VpiColumnNo(), param->VpiLineNo(), param->VpiColumnNo());
-               // its->VpiParent((any*) param->Lhs());
-                result = its;
-              } else {
-                result = (UHDM::typespec*)rhs;
+        if (component) {
+          UHDM::VectorOfparam_assign* param_assigns =
+              component->getParam_assigns();
+          if (param_assigns) {
+            for (param_assign* param : *param_assigns) {
+              const std::string& param_name = param->Lhs()->VpiName();
+              if (param_name == typeName) {
+                const any* rhs = param->Rhs();
+                if (const expr* exp = dynamic_cast<const expr*>(rhs)) {
+                  int_typespec* its = buildIntTypespec(
+                      compileDesign, param->VpiFile(), typeName,
+                      exp->VpiValue(), param->VpiLineNo(), param->VpiColumnNo(),
+                      param->VpiLineNo(), param->VpiColumnNo());
+                  // its->VpiParent((any*) param->Lhs());
+                  result = its;
+                } else {
+                  result = (UHDM::typespec*)rhs;
+                }
+                break;
               }
-              break;
             }
           }
         }

